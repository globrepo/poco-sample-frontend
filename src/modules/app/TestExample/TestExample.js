import React from "react"
import {Back, ExampleSection} from "../styled-components"
// import Styled from "./styled-components"

function TestExample () {
    return (
        <div>
            <Back to="/">Back</Back>
            <h2>Testing</h2>
            <ExampleSection>
                <div>Using snapshots</div>
                <code>expect(shallow(&#60;Component /&#62;)).toMatchSnapshot()</code>
            </ExampleSection>
            <ExampleSection>
                <div>Updating snapshots</div>
                <code>yarn test -u</code>
            </ExampleSection>
        </div>
    );
}

export default TestExample
