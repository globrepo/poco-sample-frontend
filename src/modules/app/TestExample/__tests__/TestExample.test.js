import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import TestExample from '../';

describe('<TestExample />', () => {
    const component = <TestExample />
    const wrapper= shallow(component);

    describe('Markup should stay consistent', () => {
        it('should not have any changes without a new snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('Component works', () => {
        it('should render without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(component, div);
        });
    });
});
