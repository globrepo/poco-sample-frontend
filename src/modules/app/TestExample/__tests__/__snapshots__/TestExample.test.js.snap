// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestExample /> Markup should stay consistent should not have any changes without a new snapshot 1`] = `
<div>
  <styled-components__Back
    to="/"
  >
    Back
  </styled-components__Back>
  <h2>
    Testing
  </h2>
  <styled-components__ExampleSection>
    <div>
      Using snapshots
    </div>
    <code>
      expect(shallow(&lt;Component /&gt;)).toMatchSnapshot()
    </code>
  </styled-components__ExampleSection>
  <styled-components__ExampleSection>
    <div>
      Updating snapshots
    </div>
    <code>
      yarn test -u
    </code>
  </styled-components__ExampleSection>
</div>
`;
